cmd/contour/leadership.go

```diff
@@ -133,6 +135,8 @@ func newResourceLock(ctx *serveContext, clients *k8s.Clients) resourcelock.Inter
 			Identity: resourceLockID,
 		},
 	)
-	check(err)
+	if err != nil {
+		log.WithField("context", "newResourceLock").Fatalf("err: %v", err)
```

Comment:
```Go
log.WithError(err).
    WithField("name", ctx.LeaderElectionConfig.Name).
    WithField("namespace", ctx.LeaderElectionConfig.Namespace).
    WithField("identity", resourceLockID).
    Fatal("failed to create new resource lock")
```

Note that I've added the conventional `name` and `namespace` fields because this is refereeing to a Kubernetes resource, and an `identity` key to assist debug triage.

Reply:
Done